File: Fundamentos da Programação Orientada a Objetos/Exercicios_Array/Array_OO_ex01/entidades/Aluno.java
package entidades;

public class Aluno {
    private String nome;
    private int numero;
    private float[] notas;
    
    //Construtor   
    public Aluno(String nome, int numero, float[] notas) {
        this.nome = nome;
        this.numero = numero;
        this.notas = notas;
    }
    
    //Calcular a média
    public float calcularMedia() {
        float soma = 0, media;
        
        for (float nota : notas) {
            soma = (soma + nota);
        }
        
        media = (soma / notas.length);
        return media;
    }
    
    
    //Getters and Setters
    public String getNome() {
        return nome;
    }
    
    public void setNome(String nome) {
        this.nome = nome;
    }
    
    public int getNumero() {
        return numero;
    }
    
    public void setNumero(int numero) {
        this.numero = numero;
    }
    
    public float[] getNotas() {
        return notas;
    }
    
    public void setNotas(float[] notas) {
        this.notas = notas;
    }
    
    public String toString() {
        return "Nome: " + nome + ", Número: " + numero + ", Média: " + calcularMedia();
    }
}

File: Programação Web Front-End/Lista_Array_POO/exerc_02/js/Aluno.js
export default class Aluno{
    #nome;
    #numeroMatricula;
    #notas;

    constructor(nome="", numeroMatricula="") {
        this.#nome = nome;
        this.#numeroMatricula = numeroMatricula;
        this.#notas = [];
    }

    set setNome(nome){
        this.#nome = nome;
    }
    get getNumeroMatricula(){
        return this.#numeroMatricula;
    }
    set setNumeroMatricula(numeroMatricula){
        this.#numeroMatricula = numeroMatricula;
    }
    get getNome(){
        return this.#nome;
    }

    get getNotas() {
        return this.#notas;
    }

    adicionarNota(nota) {
        this.#notas.push(nota);
    }

    calcularMedia() {
        if (this.#notas.length === 0) {
            return 0;
        }

        let soma = this.#notas.reduce(function(acumulador, nota) { 
            return (acumulador + nota);
        }, 0);

        let media = (soma / this.#notas.length);
        return media;
    }

    toString(){
        return `Aluno: \n Nome: ${this.#nome}, \n Número de Matricula: ${this.#numeroMatricula}`;
    }
}